packages/block-editor/src/components/block-fallback-web-version/index.native.js

```diff
@@ -0,0 +1,207 @@
+			if ( blockDetails.blockName === 'core/missing' ) {
+				blockDetails.blockName = blockAttributes.originalName;
+				blockDetails.blockContent =
+					blockDetails.blockName === 'core/freeform'
+						? blockAttributes.content
+						: block?.originalContent;
+			}
+
+			return {
+				isUnsupportedBlockEditorSupported:
+					capabilities?.unsupportedBlockEditor === true,
+				canEnableUnsupportedBlockEditor:
+					capabilities?.canEnableUnsupportedBlockEditor === true,
+				isEditableInUnsupportedBlockEditor:
+					! UBE_INCOMPATIBLE_BLOCKS.includes(
+						blockDetails.blockName
+					),
+				...blockDetails,
+			};
+		},
+		[ clientId ]
+	);
+
+	// Styles
+	const textStyle = usePreferredColorSchemeStyle(
+		styles.sheet__text,
+		styles[ 'sheet__text--dark' ]
+	);
+	const titleStyle = usePreferredColorSchemeStyle(
+		styles.sheet__title,
+		styles[ 'sheet__title--dark' ]
+	);
+	const descriptionStyle = usePreferredColorSchemeStyle(
+		styles.sheet__description,
+		styles[ 'sheet__description--dark' ]
+	);
+	const iconStyle = usePreferredColorSchemeStyle(
+		styles.sheet__icon,
+		styles[ 'sheet__icon--dark' ]
+	);
+	const actionButtonStyle = usePreferredColorSchemeStyle(
+		styles[ 'sheet__action-button' ],
+		styles[ 'sheet__action-button--dark' ]
+	);
+
+	const blockTitle =
+		customBlockTitle || getBlockType( blockName )?.title || blockName;
+
+	const requestFallback = useCallback( () => {
+		if (
+			canEnableUnsupportedBlockEditor &&
+			isUnsupportedBlockEditorSupported === false
+		) {
+			onCloseSheet();
+			setSendButtonPressMessage( true );
+		} else {
+			onCloseSheet();
+			setSendFallbackMessage( true );
+		}
+	}, [
+		canEnableUnsupportedBlockEditor,
+		isUnsupportedBlockEditorSupported,
+		onCloseSheet,
+	] );
+
+	const canUseWebEditor =
+		( isUnsupportedBlockEditorSupported ||
+			canEnableUnsupportedBlockEditor ) &&
+		isEditableInUnsupportedBlockEditor;
+	const actions = [
+		...[
+			canUseWebEditor && {
+				label: actionButtonLabel || __( 'Edit using web editor' ),
+				onPress: requestFallback,
+			},
+		],
+		...customActions,
+	].filter( Boolean );
+
+	return (
+		<BottomSheet
+			isVisible={ showSheet }
+			hideHeader
+			onClose={ onCloseSheet }
+			onModalHide={ () => {
+				if ( sendFallbackMessage ) {
+					// On iOS, onModalHide is called when the controller is still part of the hierarchy.
+					// A small delay will ensure that the controller has already been removed.
+					this.timeout = setTimeout( () => {
+						// For the Classic block, the content is kept in the `content` attribute.
+						requestUnsupportedBlockFallback(
+							blockContent,
+							clientId,
+							blockName,
+							blockTitle
+						);
+					}, 100 );
+					setSendFallbackMessage( false );
+				} else if ( sendButtonPressMessage ) {
+					this.timeout = setTimeout( () => {
+						sendActionButtonPressedAction(
+							actionButtons.missingBlockAlertActionButton
+						);
```

Comment:
When is this logic utilized? I'm struggling to follow what the outcome of this logic. 

My perception is that this is preexisting code copied from the Unsupported Block Editor sheet.

Reply:
Good question, I agree the action's name is not very descriptive 😅. This action is used when UBE can be used to edit the block but it's disabled for some reason. Usually, this happens in self-hosted sites connected to Jetpack that don't have the `Login with WordPress.com` option enabled. In this case, the user is prompted to enable this option and re-open the web editor again.

I checked the UBE flow in a self-hosted site with Jetpack and noticed we'd need to make some adjustments. I'll work a on fix for this 🔧 .

Reply:
Here is an example of the flow for the above case:


https://github.com/WordPress/gutenberg/assets/14905380/39950ac0-804e-4919-b8b2-53b19604eaa9